mlflow/sklearn/__init__.py

```diff
@@ -932,9 +1390,18 @@ def patched_fit(original, self, *args, **kwargs):
         )
     ]
 
+    def should_patch_class_method(class_def, func_name):
+        try:
+            if hasattr(class_def, func_name):
+                method = getattr(class_def, func_name)
+                return callable(method) and not _has_decorator(method, 'if_delegate_has_method')
```

Comment:
I found, for model methods, if it is decorated by "if_delegate_has_method" decorator, then after mlflow safe_patch it, it won't work.

We can simple verify this by:
```
import sklearn.preprocessing
import sklearn.svm
import sklearn.pipeline
estimators = [
    ("std_scaler", sklearn.preprocessing.StandardScaler()),
    ("svc", sklearn.svm.SVC()),
]
model = sklearn.pipeline.Pipeline(estimators)
X, y = get_iris()
model.fit(X, y)

# Pipeline.score is decorated by "if_delegate_has_method" decorator
# we call it in the way `cls.method(instance, *arglist, **kwargs)` which is the same way our 
# safe patching calling original fn.
# then it will raise error like `AttributeError: 'NoneType' object has no attribute '_iter'`
sklearn.pipeline.Pipeline.score(model, X, y)  
```

So this is a workaround here to skip patching these methods.

Reply:
The following estimators are affected by the issue:

(1) `transform` methods in following estimators:
```
GridSearchCV
HalvingGridSearchCV
HalvingRandomSearchCV
RandomizedSearchCV
```

(2) `predict` method in following estimators:
```
GridSearchCV
HalvingGridSearchCV
HalvingRandomSearchCV
Pipeline
RFE
RFECV
RandomizedSearchCV
SelfTrainingClassifier
StackingClassifier
StackingRegressor
```

(3) `score` method in following estimators:
```
RFE
RFECV
SelfTrainingClassifier
```

Reply:
CC @dbczumar @harupy

Reply:
@WeichenXu123 

In https://github.com/mlflow/mlflow/pull/4491/files#r676610741,

1. Can you clarify which line throws?
2. Can you include the full error stack trace in the comment?